staging/src/k8s.io/sample-controller/artifacts/examples/main.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//This is an example of polymorphic scale client.
+
+package main
+
+import (
+	"flag"
+	"os"
+
+	"github.com/golang/glog"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/clientcmd"
+	clientset "k8s.io/sample-controller/pkg/client/clientset/versioned"
+)
+
+const (
+	namespace       = "default"
+	resourceName    = "example-foo"
+	resourceGroup   = "samplecontroller.k8s.io"
+	resourceVersion = "v1alpha1"
+)
+
+func main() {
+	err := start()
+	if err != nil {
+		glog.Fatalf("error: %v", err)
+		os.Exit(1)
+	}
+
+}
+func start() error {
+	var cfg *string
+	if c := kubeconfig(); c != "" {
+		cfg = flag.String("kubeconfig", c, "(optional) KUBECONFIG env of the kubeconfig file")
+	} else {
+		cfg = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	config, err := clientcmd.BuildConfigFromFlags("", *cfg)
+	exampleClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		glog.Fatalf("error build example clientset: %s", err.Error())
+		return err
+	}
+	_, err = exampleClient.SamplecontrollerV1alpha1().Foos(namespace).Get(resourceName, metav1.GetOptions{})
+	if err != nil {
+		glog.Fatalf("err get example-foo: %s", err.Error())
+		return err
+	}
+	fooScaleClient, err := createNewScaleClient(config)
+	if err != nil {
+		glog.Fatalf("error build scale client: %s", err.Error())
+	}
+	groupResource := schema.GroupResource{
+		Group:    resourceGroup,
+		Resource: "foos",
+	}
+	gottenScale, err := fooScaleClient.Scales(namespace).Get(groupResource, resourceName)
+	if err != nil {
+		glog.Fatalf("error get the scale")
+		return nil
+	}
+	glog.Infof("Got the scale subresource: %v", gottenScale)
+	currentNo := gottenScale.Spec.Replicas
+	gottenScale.Spec.Replicas = currentNo + 1
+	fooScaleClient.Scales(namespace).Update(groupResource, gottenScale)
+	glog.Infof("Scale the replica from %v to %v using scale subresource", currentNo, gottenScale.Spec.Replicas)
+	return nil
+}
+
+func kubeconfig() string {
+	if h := os.Getenv("KUBECONFIG"); len(h) > 0 {
+		return h
+	}
+	return ""
+}
+
+func createNewScaleClient(config *rest.Config) (scale.ScalesGetter, error) {
```

Comment:
Can we get a comment explaining what this function does (i.e. this constructs a kubernetes clientset that can be used against any resource that implements the 'scale' subresource. It is special because it accepts a groupResource as part of its REST functions)